thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/pinot/PinotThirdEyeDataSource.java

```diff
@@ -246,7 +246,12 @@ public ThirdEyeResultSetGroup executePQL(PinotQuery pinotQuery) throws Execution
     Preconditions
         .checkNotNull(this.pinotResponseCache, "{} doesn't connect to Pinot or cache is not initialized.", getName());
 
-    return this.pinotResponseCache.get(pinotQuery);
+    try {
+      return this.pinotResponseCache.get(pinotQuery);
+    } catch (ExecutionException e) {
+      LOG.error("Failed to execute PQL: {}", pinotQuery.getPql());
```

Comment:
shouldn't this be caught and logged at the highest level?

Reply:
It should; however, this is the only place that has the raw PQL, which could be used on Pinot Data Explore for debugging purpose. After logging the PQL, this method rethrows the exception in order to allow higher level methods to handle the exception.